Run kind ipv6 conformance serial tests on presubmits

aojea opened the pull request:
Running ipv6 conformance serial tests on presubmits implies the job takes more than 1.30 hour.
The signal given by serial tests is not worth the time it takes, however we need signal on serial.
This PR add a new optional job to run conformance ipv6 serial tests as presubmits

aojea commented:
/assign @BenTheElder 

Also, this is not covering https://github.com/kubernetes/kubernetes/pull/86389

Will open an issue to discuss if it should be allowed in the API server to communicate in both IP families

alvaroaleman commented:
Just as an idea, would it maybe make sense to have an optional presubmit for serial ipv6 conformance tests, that ppl can manually trigger on changes like https://github.com/kubernetes/kubernetes/pull/8638 that potentially break it?

aojea commented:
> Just as an idea, would it maybe make sense to have an optional presubmit for serial ipv6 conformance tests, that ppl can manually trigger on changes like [kubernetes/kubernetes#8638](https://github.com/kubernetes/kubernetes/pull/8638) that potentially break it?

+1

/hold

let me add the new job

aojea commented:
/hold cancel

I've added 2 jobs and renamed them to match their functionality correct

* pull-kubernetes-conformance-kind-ipv6-parallel: runs the (conformance,-serial) on changes on the `test/` folder of kubernetes
* pull-kubernetes-conformance-kind-ipv6:  is optional and runs the full-conformance suite

/assign @BenTheElder

alvaroaleman commented on config/jobs/kubernetes/sig-testing/conformance-e2e.yaml:
```diff
@@ -47,6 +47,63 @@ presubmits:
+        - name: "IP_FAMILY"
+          value: "ipv6"
+        args:
+        - "--job=$(JOB_NAME)"
```
Nit: Is there a reason for not using [pod utilities](https://github.com/kubernetes/test-infra/blob/master/prow/pod-utilities.md) (`decorate: true`)? [`bootstrap.py` is deprecated](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/86389/pull-kubernetes-kubemark-e2e-gce-big/1207577736780976128/#1:build-log.txt%3A2)

aojea replied:
I've copy pasted, not familiar with `decorate`, can you link me an example so I can modify the jobs accordingly?

aojea replied:
it seems that all the e2e KIND jobs are using bootstrap.py, but the new ones, but those are using a different container image.
I think that it will be better to do the migration to podutilities of all the KIND jobs in a follow up PR, to have consistency, what do you think @BenTheElder ?

dims replied:
@aojea i have some notes that may help - https://gist.github.com/dims/c1296f8ed42238baea0a5fcae45f4cf4#file-notes-md

BenTheElder replied:
> I think that it will be better to do the migration to podutilities of all the KIND jobs in a follow up PR, to have consistency, what do you think @BenTheElder ?

yes, please. let's minimize the diffs that are not related.

> Nit: Is there a reason for not using pod utilities (decorate: true)? bootstrap.py is deprecated

nearly all jobs are because nobody is signing up to make this migration.

we also still have bugs blocking data collection working effectively so i'm hesitant to encourage moving.

BenTheElder replied:
I've experimentally migrated a more recent job to a much more streamlined podutils config + image but I do not think we should migrate the older kubernetes release jobs @aojea, they will be phased out. we should avoid touching jobs that are not for master.

BenTheElder commented:
> Run kind ipv6 conformance serial tests on presubmits #15655

this PR title is pretty misleading, we're _optionally_ running them.

BenTheElder commented on config/jobs/kubernetes/sig-testing/conformance-e2e.yaml:
```diff
@@ -47,6 +47,63 @@ presubmits:
+    - master
+    always_run: false
+    skip_report: false
+    max_concurrency: 8
```
we may need to reduce these with more jobs and not more infra ...

aojea replied:
copy paste :)

BenTheElder commented:
/lgtm
/approve
be aware that we don't have more ipv6 nodes and we probably won't be expanding that pool

I will again bringup switching the main pool to ubuntu in the new year, but right now the oncall team is mostly out.
